src/main/java/com/puppycrawl/tools/checkstyle/xpath/XpathQueryGenerator.java

```diff
@@ -75,6 +76,11 @@
  */
 public class XpathQueryGenerator {
 
+    /**
+     * This regexp is used to convert new line to newline tag.
+     */
+    private static final Pattern NEWLINE_TO_TAG = Pattern.compile("[\n]+");
```

Comment:
What about other non-printable characters, like `\r` for Windows which does `\r\n`?

Reply:
Done.

Reply:
You are now changing any `\r` to `\n`. I am not sure if this is desired.

I was under the impression we should change `\r` to `\\r` basically.

Reply:
@AkMo3 , please keep creating separate test input for each problem we point. UT should prove good behavior on all cases.

Reply:
Added tests for 3 types. Please check.